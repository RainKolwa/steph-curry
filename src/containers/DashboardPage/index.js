import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Menu, UserHead, RecommendItem } from '../../components'
import TaskBoard from '../TaskBoard'
import PostList from '../PostList'
import './style.styl'

export class DashboardPage extends Component {
  render() {
    const { user } = this.props
    return (
      <div className="page-dashboard">
        <Menu />
        {user && <UserHead user={user} />}
        <TaskBoard limit={3} />
        <PostList path="inbox" infinite>
          <RecommendItem />
        </PostList>
      </div>
    )
  }
}

export default connect(state => {
  const { user } = state.auth
  return {
    user,
  }
})(DashboardPage)
