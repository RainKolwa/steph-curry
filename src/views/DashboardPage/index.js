import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'
import { Menu, UserHead, RecommendItem } from '../../components'
import TaskBoard from '../../containers/TaskBoard'
import PostList from '../../containers/PostList'
import './style.styl'

export class DashboardPage extends Component {
  render() {
    const { user } = this.props
    return (
      <div className="page-dashboard">
        <Helmet>
          <title>学习计划</title>
        </Helmet>
        <Menu />
        {user && <UserHead user={user} />}
        <TaskBoard limit={3} />
        <PostList path="inbox" infinite>
          <RecommendItem />
        </PostList>
      </div>
    )
  }
}

export default connect(state => {
  const { user } = state.auth
  return {
    user,
  }
})(DashboardPage)
