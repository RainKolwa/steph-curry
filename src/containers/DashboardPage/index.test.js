import React from 'react'
import { mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import DashboardPage from './index'

const mockStore = configureStore()

describe('<DashboardPage />', () => {
  const getState = { auth: {} }
  const store = mockStore(getState)
  it('should render div', () => {
    const wrapper = mount(<DashboardPage store={store} />)
    expect(wrapper.find('div')).toHaveLength(1)
  })
})
