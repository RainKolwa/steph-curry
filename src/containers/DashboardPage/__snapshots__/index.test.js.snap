exports[`<DashboardPage /> renders correctly 1`] = `
<div>
  <ul
    className="menu-main">
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        学习
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        课程
      </a>
    </li>
    <li>
      <a
        onClick={[Function]}
        style={Object {}}>
        作业
      </a>
    </li>
  </ul>
</div>
`;
